geospaas_harvesting/verify.py

```diff
@@ -0,0 +1,51 @@
+""" Verification module. """
+import os
+from datetime import datetime
+
+import django
+import requests
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_harvesting.settings')
+django.setup()
+
+from geospaas.catalog.models import DatasetURI
+
+def main():
+    """ Verifies the datasets based on their dataseturi. If the download link does not provide a
+    download availability and returns a text or html response, then the dataset uri is removed. If
+    there is no other "dataseturi" remains for the dataset, then the dataset is also removed in
+    order to be harvested again in the future with a correct and healthy "dataseturi".
+    Since the number of datasets in the database might be enormous, the datasets are retrieved
+    into with a variable named retrieved_dataset_uris with an specific length number
+    for memory management. """
+    corrupted_url_set = set()
+    id_range = range(DatasetURI.objects.earliest('id').id,
+                     DatasetURI.objects.latest('id').id, 1000)# <=number for the length of retrieved
+    for i in range(len(id_range)):
+        try:
+            retrieved_dataset_uris = DatasetURI.objects.filter(
+                id__gte=id_range[i], id__lt=id_range[i+1])
+        except IndexError:
+            retrieved_dataset_uris = DatasetURI.objects.filter(
+                id__gte=id_range[i], id__lte=DatasetURI.objects.latest('id').id)
+        for dsuri in retrieved_dataset_uris:
+            content_type = requests.head(dsuri.uri, allow_redirects=True).headers.get('content-type')
+            if 'html' in content_type.lower() or 'text' in content_type.lower():
+                corrupted_url_set.add(dsuri.uri)
+
+    with open(f"unverified_ones_at_{datetime.now().strftime('%Y-%m-%d|%H_%M_%S')}.txt", 'w') as f:
```

Comment:
- Please use a clear descriptive name: "ones" could be anything. Keep this in mind for naming in general.
- It's a very bad idea to use `|` in a file name since it's the pipe operator in Unix shells.
- Either of these solutions would be more user-friendly:
    - the name of the file could be passed as an argument to the script (with a default value if it is not provided).
    - the script could write its output to stdout, and the user can then redirect it as they wish.

Reply:
Done in the next commit.